torchgen/model.py

```diff
@@ -933,7 +962,10 @@ def get_kernel(
             f = self.primary(g)
         else:
             assert_never(f)
-        if f.func.name not in self.index:
+        if self.should_gen_dispatchless_composite(f):
+            # Generate a 'BackendMetadata' for dispatch-less composite kernels.
+            return BackendMetadata(self.dispatchless_composite_kernel(f), False)
```

Comment:
Tbh it would be nice if these were all added directly into `self.index` up front, that way the object itself is a source of truth for "what ops does this backend support" (instead of being required to indirect through this function).

If it's annoying to do that though then I think this isn't too big of a deal

Reply:
might also be worth calling out explicitly - the `False` here is because generated composite kernel logic in this PR only applies to non-structured operators.

Reply:
I thought about doing that. But, to make it consistent, I would have to add the non-out overloads of structured kernels. And, that, could change how other parts of the codegen work. I guess we would better do it in another PR.

> might also be worth calling out explicitly - the `False` here is because generated composite kernel logic in this PR only applies to non-structured operators.

Not sure what you mean here. Could you elaborate?

Reply:
> might also be worth calling out explicitly - the False here is because generated composite kernel logic in this PR only applies to non-structured operators.

Oh nvm, I guess this doesn't matter too much.
 
`BackendMetadata` has a second field `structured`, which indicates whether or not the given *kernel* is structured (for that operator + backend). Which isn't too relevant to the code you're changing - since the generated composite kernels you're generating never go through structured kernel codegen, it seems fine for this to always be False.